cips/cip-pfb-limits.md

```diff
@@ -0,0 +1,44 @@
+| cip | TBD(27) |
+| - | - |
+| title | Block limits for number of PFBs and non-PFBs |
+| description | Set limits for number of PFBs and non-PFBs per block |
+| author | Josh Stein ([@jcstein](https://github.com/jcstein)), Nina Barbakadze ([@ninabarbakadze](https://github.com/ninabarbakadze)), rach-id ([@rach-id](https://github.com/rach-id)), Rootul Patel ([@rootulp](https://github.com/rootulp)) |
+| discussions-to | <https://forum.celestia.org/t/cip-limit-number-of-pfbs-and-non-pfbs-per-block-increase-transaction-size-limit/1843> |
+| status | Draft |
+| type | Standards Track |
+| category | Core |
+| created | 2024-10-16 |
+
+## Abstract
+
+This CIP proposes to set limits for the number of PayForBlobs (PFBs) messages and non-PFBs messages per block. The proposal is to set the limits to 600 PFBs messages and 200 non-PFBs messages per block. Setting PFB and non-PFBs limits is not consensus-breaking.
+
+## Specification
+
+1. The number of PFBs per block is limited to 600 by setting `MaxPFBMessages` to 600.
+1. The number of non-PFBs messages per block is limited to 200 by setting `MaxNonPFBMessages` to 200.
+1. It's important to note that these limits are not strictly enforced. While they are defined by the `celestia-app` implementation, a validator could potentially modify the `PrepareProposal` logic, run a custom binary, and produce blocks that exceed the specified limits for PFB or non-PFBs transactions.
+
+## Rationale
+
+The rationale for this proposal is to prevent long block times on the network by limiting the number of PFBs and non-PFBs messages per block. This was initially not considered consensus-breaking, but it has a meaningful effect on users and should be formalized in a CIP.
```

Comment:
```suggestion
The rationale for this proposal is to prevent long block times on the network by limiting the number of PFBs and non-PFB messages per block. This is not consensus-breaking but it has a meaningful effect on users and should be formalized in a CIP.
```